bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AbstractCommandProvider.java

```diff
@@ -150,25 +150,31 @@ public void removedService(ServiceReference reference, Object service) {
      * This method is responsible for execution of the {@link AutomationCommandExport} operation by choosing the
      * {@link Parser} which to be used for exporting a set of automation objects, in a file. When the choice is made,
      * the chosen {@link Parser} is used to do the export.
-     *
+     * 
      * @param parserType is a criteria for choosing the {@link Parser} which to be used.
      * @param set a Set of automation objects for export.
      * @param file is the file in which to export the automation objects.
      * @throws Exception is thrown when I/O operation has failed or has been interrupted or generating of the text fails
      *             for some reasons.
      */
-    public void exportData(String parserType, Set<E> set, File file) throws Exception {
-        OutputStreamWriter oWriter = null;
-        oWriter = new OutputStreamWriter(new FileOutputStream(file));
+    @SuppressWarnings("resource")
+    public String exportData(String parserType, Set<E> set, File file) throws Exception {
+        OutputStreamWriter oWriter = new OutputStreamWriter(new FileOutputStream(file));
```

Comment:
Shouldn't we use try-with-resources instead of the suppress warnings annotation?